File: source/windowsXP/NetLayer/CommHandlerI.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace NetLayer
{
    /// <summary>
    /// Interfaz para eventos de comunicación generados en NetHandler
    /// </summary>
    public interface CommHandlerI
    {
        /// <summary>
        /// Se gatilla cuando debe comenzar la comunicación
        /// </summary>
        void startNetworkingHandler();
        /// <summary>
        /// Se gatilla cuando se debe detener la comunicación 
        /// </summary>
        void stopNetworkingHandler();
        /// <summary>
        /// Se gatilla cuando se reseteará la coneción
        /// </summary>
        void resetNetworkingHandler();
        /// <summary>
        /// Se gatilla cuando ocurre un error en la red
        /// </summary>
        /// <param name="e">La excepción generada</param>
        void errorNetworkingHandler(Exception e);
        /// <summary>
        /// Se gatilla para enviar información del estado de la red.
        /// </summary>
        /// <param name="message">El mensaje envíado</param>
        void informationNetworkingHandler(String message);
    }
}

File: source/java/hlmp/NetLayer/CommHandlerI.java
package hlmp.NetLayer;

public interface CommHandlerI {
	
	/**
	 * Se gatilla cuando debe comenzar la comunicación
	 */
    abstract void startNetworkingHandler();
    
    /**
     * Se gatilla cuando se debe detener la comunicación
     */
    abstract void stopNetworkingHandler();
    
    /**
     * Se gatilla cuando se reseteará la coneción
     */
    abstract void resetNetworkingHandler();
    
    /**
     * Se gatilla cuando ocurre un error en la red
     * @param e La excepción generada
     */
    abstract void errorNetworkingHandler(Exception e);
    
    /**
     * Se gatilla para enviar información del estado de la red.
     * @param message El mensaje envíado
     */
    abstract void informationNetworkingHandler(String message);

}
